<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="David Xin Zhao" />


<title>Multivariable regression of microbiome and IBD using MaAsLin2 R package</title>

<script src="site_libs/header-attrs-2.18/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="styles.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">My Website</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="my-website.html">Home</a>
</li>
<li>
  <a href="index.html">Multivariable regression</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Multivariable regression of microbiome and
IBD using MaAsLin2 R package</h1>
<h4 class="author">David Xin Zhao</h4>
<h4 class="date">Last edited 16 November 2022</h4>

</div>


<p>knit: (function(input_file, encoding) { out_dir &lt;- ‘docs’;
rmarkdown::render(input_file, encoding=encoding,
output_file=file.path(dirname(input_file), out_dir, ‘index.html’))})</p>
<style>
#TOC{
 color: #708090;
 font-family: Calibri;
 font-size: 16px;
 border-color: #708090;
}
h1.title{
 color: #800000;
 background-color: #F5F5F5;
 opacity: 0.6;
 font-family: Calibri;
 font-size: 40px;
}
h4.author{
 color: #708090;
 font-family: Calibri;
 font-size: 40px;
}
h4.date{
 color: #708090;
 font-family:Calibri; 
}
body{
 color: #708090; font-family: Calibri; background-color: #F5F5F5;
}
pre{
 color: #708090;
 background-color: #F8F8FF;
}
</style>
<div id="maaslin2" class="section level2">
<h2>MaAsLin2</h2>
<p><a
href="https://github.com/biobakery/biobakery/wiki/maaslin2">MaAsLin2</a>
is to efficiently determine multivariable association between
phenotypes, environments, exposures, covariates and microbial meta-omics
features.</p>
</div>
<div id="install-maaslin2-with-bioconductor" class="section level2">
<h2>Install MaAsLin2 with Bioconductor</h2>
<pre class="r"><code># if(!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
#     install.packages(&quot;BiocManager&quot;)
# BiocManager::install(&quot;Maaslin2&quot;)

library(Maaslin2)
library(tidyverse)</code></pre>
</div>
<div
id="multiple-regression-of-microbiome-and-clincial-covariates-of-interest"
class="section level2">
<h2>Multiple regression of microbiome and clincial covariates of
interest</h2>
<div
id="load-data-sets-use-example-datasets-provided-by-maaslin2-package"
class="section level3">
<h3>Load data sets (use example datasets provided by MaAsLin2
package)</h3>
<pre class="r"><code> 
# microbiome abundance data 
df_input_data &lt;- read.table(file = input_data,
                            header = TRUE,
                            sep = &quot;\t&quot;,
                            row.names = 1,
                            stringsAsFactors = FALSE) 

# meta data 
df_input_metadata &lt;- read.table(file=input_metadata,
                                header = T,
                                sep=&quot;\t&quot;,
                                row.names = 1,
                                stringsAsFactors = F) 

# pathway data 
df_input_path &lt;- data.frame(read.csv(&quot;./pathabundance_relab.tsv&quot;, 
                                     sep = &quot;\t&quot;, 
                                     stringsAsFactors = F, 
                                     row.names = 1))</code></pre>
</div>
<div id="fit-mutiple-linear-regression-model-on-microbiome-data"
class="section level3">
<h3>Fit mutiple linear regression model on microbiome data</h3>
<p>The following command runs MaAsLin2 on the HMP2 data, running a
multivariable regression model to test for the association between
microbial species abundance versus IBD diagnosis and dysbiosis
scores.</p>
<pre class="r"><code>fit_data2 &lt;- Maaslin2(
        input_data = df_input_data,
        input_metadata = df_input_metadata,
        min_prevalence = 0, #without additional filtering 
        normalization = &quot;NONE&quot;,  #without additional normalization  
        output = &quot;demo_output2&quot;,
        fixed_effects = c(&quot;diagnosis&quot;, &quot;dysbiosis&quot;),  #set diagnosis and dysbiosis as fixed effect variables 
        reference = c(&quot;diagnosis,nonIBD&quot;)) #set the baseline for the categorical variable, diagnosis 
## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
all_results &lt;- fit_data2$results #extract coefficients and q-value</code></pre>
</div>
<div id="evaluate-significant-genus-based-on-model-parameters"
class="section level3">
<h3>Evaluate significant genus based on model parameters</h3>
<p>significant associations is provided in ‘significant_results.tsv’</p>
<pre class="r"><code>sig_results &lt;- all_results %&gt;% 
        filter(qval &lt;= 0.05) %&gt;% #sig threshold 
        arrange(qval)

head(sig_results, 5) #view first rows </code></pre>
<pre><code>##                        feature  metadata     value      coef     stderr          pval      name          qval    N
## 1 Faecalibacterium.prausnitzii dysbiosis dysbiosis -2.647940 0.07465408 1.544980e-203 dysbiosis 4.032398e-201 1595
## 2        Bacteroides.uniformis dysbiosis dysbiosis -3.503693 0.11995326 1.589498e-150 dysbiosis 2.074294e-148 1595
## 3 Subdoligranulum.unclassified dysbiosis dysbiosis -1.525402 0.06188741 6.962634e-114 dysbiosis 6.057492e-112 1595
## 4         Alistipes.putredinis dysbiosis dysbiosis -3.444523 0.14190247 5.443288e-111 dysbiosis 3.551745e-109 1595
## 5           Ruminococcus.obeum dysbiosis dysbiosis -1.946980 0.08362614 2.023721e-103 dysbiosis 1.056382e-101 1595
##   N.not.zero
## 1       1534
## 2       1391
## 3       1546
## 4        940
## 5       1346</code></pre>
</div>
<div id="fit-multivariable-regression-models-on-functional-data"
class="section level3">
<h3>Fit multivariable regression models on functional data</h3>
<p>Authors tend to work with CPM units because we find them to be more
convenient, but they are numerically equivalent to relative abundances
for modeling purposes (CPM = RA * 1e6)</p>
<p>Quick tips for using MaAsLin2 on functional profiles: 1.To reduce the
number of features, you typically want to run either the unstratified
functional features ( or a filtered subset of the stratified features)
2.You may also want to remove functional features that are highly
correlated to one specific taxon (i.e. likely contributed by that
microbe), since these can be better-explained by taxonomic changes</p>
<pre class="r"><code>#This can also be done with with the HUMAnN 3 untiliy `humann_split_stratified_table`
unstrat_pathways &lt;-function(dat_path){
  temp = dat_path[!grepl(&quot;\\|&quot;,rownames(dat_path)),]
  return(temp)
}

df_input_path = unstrat_pathways(df_input_path)</code></pre>
<p>Run the same model as above on the MetaCyc pathway table generated
from the bioBakery workflows.</p>
<pre class="r"><code># fit the model on pathway (functional) data - it takes a few minutes  
fit_func &lt;- Maaslin2(
        input_data = df_input_path,
        input_metadata = df_input_metadata,
        output = &quot;demo_functional&quot;,
        fixed_effects = c(&quot;diagnosis&quot;, &quot;dysbiosis&quot;),
        reference=c(&quot;diagnosis,nonIBD&quot;),
        min_abundance = 0.0001,
        min_prevalence = 0.1) 
## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
# note: here we have the samples in the columns and the features in the rows and MaAsLin2 correctly identified this and was able to match the samples. 

all_results_func &lt;- fit_func$results</code></pre>
</div>
<div id="interactions" class="section level3">
<h3>Interactions</h3>
<p>Unfortunately, MaAsLin2 does not yet provide a direct interface for
testing for interaction. Instead, the user needs to create artificial
interaction columns as additional ‘fixed_effects’ terms.</p>
<p>Using the above fit as an example, to test for the interaction
between ‘diagnosis’ and ‘dysbiosis’, I can create two additional
columns: ‘CD_dysbiosis’ and ‘UC_dysbiosis’ (since the reference for
‘diagnosis’ is ‘nonIBD’):</p>
<pre class="r"><code>
df_input_metadata$CD_dysbiosis &lt;- (df_input_metadata$diagnosis == &quot;CD&quot;) * df_input_metadata$dysbiosis

df_input_metadata$UC_dysbiosis &lt;- (df_input_metadata$diagnosis == &quot;UC&quot;) * df_input_metadata$dysbiosis

fit_data_interaction &lt;- Maaslin2(
        input_data = df_input_data,
        input_metadata = df_input_metadata,
        min_prevalence = 0,
        normalization = &quot;NONE&quot;,
        output = &quot;demo_output_interaction&quot;,
        fixed_effects = c(&quot;diagnosis&quot;, &quot;dysbiosis&quot;, &quot;CD_dysbiosis&quot;, &quot;UC_dysbiosis&quot;),
        reference = c(&quot;diagnosis,nonIBD&quot;)) 
## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</code></pre>
<p><img src="index_files/figure-html/interaction%20term-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
all_results_interaction &lt;- fit_data_interaction$results  # store results </code></pre>
</div>
<div id="random-effects" class="section level3">
<h3>Random effects</h3>
<p>We note that HMP2 is a longitudinal design where the same subject
(column ‘subject’) can have multiple samples. We thus ask MaAsLin2 to
use subject as its random effect grouping variable:</p>
<pre class="r"><code>
fit_data_random &lt;- Maaslin2(
        input_data = df_input_data,
        input_metadata = df_input_metadata,
        min_prevalence = 0,
        normalization = &quot;NONE&quot;,
        output = &quot;demo_output_random&quot;,
        fixed_effects = c(&quot;diagnosis&quot;, &quot;dysbiosis&quot;),
        random_effects = c(&quot;subject&quot;),
        reference = c(&quot;diagnosis,nonIBD&quot;)
)
## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</code></pre>
<p><img src="index_files/figure-html/mixed%20effect%20model-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
all_results_random &lt;- fit_data_random$results</code></pre>
<p>If you are interested in testing the effect of time in a longitudinal
study, then the itme point variable should be included in
‘fixed_effects’ during your MaAsLin2 call.</p>
</div>
<div id="prevalence-and-abundance-filtering-in-maaslin2"
class="section level3">
<h3>Prevalence and abundance filtering in MaAsLin2</h3>
<p>minimum prevalence threshold: 10-50% (up to 70-90% can be reasonable)
minimum relative abundance threshold: 0.0001</p>
<pre class="r"><code>
fit_data_filter = Maaslin2(
    input_data = df_input_data, 
    input_metadata = df_input_metadata, 
    normalization = &quot;NONE&quot;,
    output = &quot;demo_output_filter&quot;, 
    fixed_effects = c(&quot;diagnosis&quot;, &quot;dysbiosis&quot;),
    reference = c(&quot;diagnosis,nonIBD&quot;),
    random_effects = c(&quot;subject&quot;),
    min_prevalence = 0.1,
    min_abundance = 0.0001)
## Warning in xtfrm.data.frame(x): cannot xtfrm data frames</code></pre>
<p><img src="index_files/figure-html/prevalence%20and%20abundance%20filtering-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
